ckan/lib/helpers.py

```diff
@@ -868,7 +868,11 @@ def dict_list_reduce(list_, key, unique=True):
     values for the key with unique values if requested. '''
     new_list = []
     for item in list_:
-        value = item.get(key)
+        try:
+            value = item.get(key)
+        except AttributeError:
+            # item might not be a dict.
+            continue
```

Comment:
Ross Jones wrote:
> As the function is documented as taking a list of dicts, it's possibly
> not the right thing to do, to just silently ignore the bad data passed
> in. Perhaps if you raise an error that points out that it has been
> passed bad data (e.g. not a list of dicts) may help in tracking down the
> real cause of the problem (possibly in a schema somewhere?).


I agree this is not the right thing to do (just did not read the 
docstring :P). I can debug this live if you point me to something to 
look for. The full traceback does not look very helpful from my point of 
view...